src/main/java/com/thebois/models/beings/AbstractBeingGroup.java

```diff
@@ -3,12 +3,22 @@
 import java.util.ArrayList;
 import java.util.Collection;
 
+import com.google.common.eventbus.Subscribe;
+
+import com.thebois.Pawntastic;
+import com.thebois.listeners.events.OnDeathEvent;
+
 /**
  * An abstract implementation of IBeingGroup.
  */
 public abstract class AbstractBeingGroup implements IBeingGroup {
 
     private Collection<IBeing> beings = new ArrayList<>();
+    private Collection<IBeing> deadBeings = new ArrayList<>();
+
+    protected AbstractBeingGroup() {
+        Pawntastic.getEventBus().register(this);
```

Comment:
Have you made sure that this event stays registered after saving/loading from a save file?